vizro-core/src/vizro/actions/_callback_mapping/_callback_mapping_utils.py

```diff
@@ -203,9 +203,9 @@ def _get_export_data_callback_outputs(action_id: ModelID) -> Dict[str, List[Stat
         targets = _get_components_with_data(action_id=action_id)
 
     return {
-        f"download-dataframe_{target}": Output(
+        f"download_dataframe_{target}": Output(
             component_id={
-                "type": "download-dataframe",
+                "type": "download_dataframe",
```

Comment:
Changed because the dictionary keys become namedtuple attributes at some point and they must not contain a hyphen character.

Reply:
I think this is a good new restriction btw, because it makes the names of outputs work similar to how we handle input names, which also need to be valid Python variable names (so we convert `.` to `_`).